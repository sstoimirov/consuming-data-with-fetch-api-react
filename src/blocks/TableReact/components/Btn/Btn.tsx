import * as React from 'react';

export class Btn extends React.PureComponent<{ text: string, onClick: (e: React.MouseEvent<HTMLButtonElement>) => void; }, {}>{
    render() {
        return (
            <button type="button" onClick={this.props.onClick}>{this.props.text}</button>
        )
    }
}