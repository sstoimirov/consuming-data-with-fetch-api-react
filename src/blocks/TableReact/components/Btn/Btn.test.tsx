import * as React from "react";
import { shallow, configure } from "enzyme";
import { Btn } from "./Btn";
import Adapter from 'enzyme-adapter-react-16'

configure({ adapter: new Adapter() })

describe("Btn", () => {
    it("Renders correctly", () => {
        const comp = shallow(<Btn text="Button" onClick={jest.fn()} />);
        expect(comp.debug()).toMatchSnapshot();
    })
})