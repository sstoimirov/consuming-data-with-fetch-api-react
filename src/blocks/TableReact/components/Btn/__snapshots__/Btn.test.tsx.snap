// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Btn Renders correctly 1`] = `
"<button type=\\"button\\" onClick={[Function: mockConstructor]}>
  Button
</button>"
`;
